### PR TITLE
fix: regression with migrate command

### LINKED ISSUE #1422: Migrate functionality doesn't work as expected
**Describe the bug**

**What is the current behavior?**
The `migrate` command doesn't work as expected, it throws an error without updating the config file.

**To Reproduce**

Type in `webpack-cli migrate <path-to-config>` 

**Expected behavior**
The config is updated as expected.

**Screenshots**
Screenshot 2020-04-06 at 12 27 56 AM

**Please paste the results of `webpack-cli info` here, and mention other relevant information**

```sh
  System:
    OS: macOS 10.15.4
    CPU: (8) x64 Intel(R) Core(TM) i5-8279U CPU @ 2.40GHz
    Memory: 624.41 MB / 8.00 GB
  Binaries:
    Node: 13.11.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
  Browsers:
    Safari: 13.1
  npmPackages:
    webpack: ^1.15.0 => 1.15.0 
  npmGlobalPackages:
    webpack-cli: 4.0.0-beta.8
```

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
The `migrate` command expects a source config file to be  passed in as an argument. And for the output, we can either have it as another file or overwrite the source. Added test cases covering these scenarios.

**If relevant, did you update the documentation?**
N/A

**Summary**
closes #1422 

**Does this PR introduce a breaking change?**
Nope

**Other information**
N/A

### CODE REVIEW

File: packages/migrate/src/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import chalk from 'chalk';
-import diff from 'diff';
+import { Change, diffLines } from 'diff';
```
Comment (jamesgeorge007) [inline packages/migrate/src/index.ts:2]: Surprisingly individual named imports seems to the fix the issue.

Comment (jamesgeorge007) [reply packages/migrate/src/index.ts:2]: Ig the issue is arising from the type definitions lib.

**Comment by jamesgeorge007** [comment]
> It is change `webpack.config.js` content after tests?

That would require us to bypass the prompts for the entire process complete. As of now, we're checking against the very first user prompt.

**Comment by evilebottnawi** [comment]
> That would require us to bypass the prompts for the entire process to complete. As of now, we're checking against the very first user prompt.

We should do it, because other steps can be broken too

File: test/migrate/config/migrate-config.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    it('should prompt for config validation when an output path is provided', async () => {
+        const { stdout } = await runAndGetWatchProc(__dirname, ['migrate', 'webpack.config.js', 'updated-webpack.config.js'], false, 'y');
+        expect(stdout).toContain('? Do you want to validate your configuration?');
+    });
+
+    it('should generate an updated config file when an output path is provided', async () => {
+        const stdout = await runPromptWithAnswers(__dirname, ['migrate', 'webpack.config.js', 'updated-webpack.config.js'], [ENTER, ENTER]);
+        expect(stdout).toContain('? Do you want to validate your configuration?');
+    });
```
Comment (jamesgeorge007) [inline test/migrate/config/migrate-config.test.js:31]: This doesn't seem to work as expected:-
```js
  ● migrate command › should generate an updated config file when an output path is provided

    expect(received).toContain(expected) // indexOf

    Expected substring: "? Do you want to validate your configuration?"
    Received string:    ""
```

Comment (anshumanv) [reply test/migrate/config/migrate-config.test.js:31]: Can you log stdout?

Comment (jamesgeorge007) [reply test/migrate/config/migrate-config.test.js:31]: It is an empty string as evident from above.

Comment (anshumanv) [reply test/migrate/config/migrate-config.test.js:31]: Ah yes saw, hm that's unusual

Comment (anshumanv) [reply test/migrate/config/migrate-config.test.js:31]: What do you see with `run`?

Comment (jamesgeorge007) [reply test/migrate/config/migrate-config.test.js:31]: Well `run` works absolutely fine as you can see from rest of the test cases.

**Comment by jamesgeorge007** [comment]
#1495

**Pull request closed without merging**